django/http/request.py

```diff
@@ -480,15 +480,22 @@ def build_request_repr(request, path_override=None, GET_override=None,
                else pformat(request.GET))
     except Exception:
         get = '<could not parse>'
+
     if request._post_parse_error:
         post = '<could not parse>'
     else:
         try:
-            post = (pformat(POST_override)
-                    if POST_override is not None
-                    else pformat(request.POST))
+            if POST_override is not None:
+                post = pformat(POST_override)
+            else:
+                # Use the exception reporter filter to return the cleansed POST data
+                from django.views.debug import get_exception_reporter_filter
```

Comment:
Sounds goood. Feel free to submit that as a separate pull request. We can probably merge that first and then rebase this.